auth/src/main/java/com/firebase/ui/auth/util/data/AuthViewModelBase.java

```diff
@@ -0,0 +1,28 @@
+package com.firebase.ui.auth.util.data;
+
+import android.app.Application;
+
+import com.firebase.ui.auth.util.ui.FlowHolder;
+import com.firebase.ui.auth.util.ui.ViewModelBase;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.PhoneAuthProvider;
+
+public class AuthViewModelBase extends ViewModelBase<FlowHolder> {
+    protected FlowHolder mFlowHolder;
+    protected FirebaseAuth mAuth;
```

Comment:
Does putting `FirebaseAuth` in the `ViewModel` leak a `Context`?

Reply:
No? I'm pretty sure `FirebaseAuth` only has access to app contexts.

Reply:
I think you're right.  It's only `FirebaseAnalytics` that takes a context when you get an instance.

Reply:
Even analytics doesn't use the context (dunno why):
![image](https://user-images.githubusercontent.com/9490724/33967554-c6c96dde-e018-11e7-94e6-dd6c4962e04e.png)
![image](https://user-images.githubusercontent.com/9490724/33967571-df8f3d44-e018-11e7-8c60-3672a3dd1b8b.png)
![image](https://user-images.githubusercontent.com/9490724/33967590-eebd52ec-e018-11e7-9beb-ea14033b6cc6.png)